src/huggingface_hub/keras_mixin.py

```diff
@@ -1,28 +1,161 @@
+import json
 import logging
 import os
 from pathlib import Path
+from typing import Any, Dict, Optional, Union
 
-from huggingface_hub import ModelHubMixin, hf_hub_download
+from huggingface_hub import ModelHubMixin
+from huggingface_hub.file_download import is_tf_available
+from huggingface_hub.snapshot_download import snapshot_download
+
+from .constants import CONFIG_NAME
+from .hf_api import HfApi, HfFolder
+from .repository import Repository
 
 
 logger = logging.getLogger(__name__)
 
+if is_tf_available():
+    import tensorflow as tf
+
+
+def save_pretrained_keras(
+    model, save_directory: str, config: Optional[Dict[str, Any]] = None
+):
+    """Saves a Keras model to save_directory in SavedModel format. Use this if you're using the Functional or Sequential APIs.
+
+    model:
+        The Keras model you'd like to save. It model must be compiled and built.
+    save_directory (:obj:`str`):
+        Specify directory in which you want to save the Keras model.
+    config (:obj:`dict`, `optional`):
+        Configuration object to be saved alongside the model weights.
+    """
+
+    if not model.built:
+        raise ValueError("Model should be built before trying to save")
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # saving config
+    if isinstance(config, dict):
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump(config, f)
```

Comment:
config is not required, so I figure we just pass here. should I add a warning/info message here instead?

Reply:
A warning/info sounds ok, but if a configuration is passed and nothing can be done out of it since it's in an invalid format (i.e., not a dict), I think it would be a better user experience to error out. If nothing is passed, then it's no problem to just do nothing!